app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt

```diff
@@ -44,18 +48,23 @@ class WallpaperSettingsFragment : Fragment() {
             setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
             setContent {
                 FirefoxTheme {
-                    var currentWallpaper by remember { mutableStateOf(wallpaperManager.currentWallpaper) }
+                    val wallpapers = appStore.observeAsComposableState { state ->
+                        state.wallpaperState.availableWallpapers
+                    }.value ?: listOf()
+                    var currentWallpaper = appStore.observeAsComposableState { state ->
+                        state.wallpaperState.currentWallpaper
+                    }.value ?: Wallpaper.Default
                     var wallpapersSwitchedByLogo by remember { mutableStateOf(settings.wallpapersSwitchedByLogoTap) }
                     WallpaperSettings(
```

Comment:
nit: could we add a newline between `wallpapersSwitchedByLogo` and `WallpaperSettings`